### ISSUE TITLE
TypeError - Cannot read properties of undefined (reading 'packageJson')

### ISSUE THREAD

**Issue by kheniparth** [bizon/selling-partner-api-sdk#545]
Happy Friday :) 

## Context 

I would like to create a private Amazon SP-API application. I have used a few (auth, catalog, pricing, fees) sub-packages from this library to get the data I would like for my application. 

This app would be something like ScoutIQ that helps me check some details that interest me. I am trying to host this application on Netlify to be served. I have used the sub-packages with the Netlify functions to make the API calls. 

Everything works fine in the dev environment, I could make actual API calls with valid data so that means my credentials are correct. 

When I deploy the application to Netlify the build succeeds but whenever I call the functions that will eventually make the API call using this library I get this error

# [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) - Cannot read properties of undefined (reading 'packageJson')

## Stack Trace by Netlify

```txt
TypeError: Cannot read properties of undefined (reading 'packageJson')
    at node_modules/@sp-api-sdk/auth/dist/cjs/src/utils/package.js (/var/task/netlify/functions/pricing.js:19601:34)
    at __require (/var/task/netlify/functions/pricing.js:37:50)
    at node_modules/@sp-api-sdk/auth/dist/cjs/src/utils/axios.js (/var/task/netlify/functions/pricing.js:19613:21)
    at __require (/var/task/netlify/functions/pricing.js:37:50)
    at node_modules/@sp-api-sdk/auth/dist/cjs/src/access-token.js (/var/task/netlify/functions/pricing.js:19630:19)
    at __require (/var/task/netlify/functions/pricing.js:37:50)
    at node_modules/@sp-api-sdk/auth/dist/cjs/src/index.js (/var/task/netlify/functions/pricing.js:30983:26)
    at __require (/var/task/netlify/functions/pricing.js:37:50)
    at node_modules/@sp-api-sdk/auth/dist/cjs/index.js (/var/task/netlify/functions/pricing.js:31073:26)
    at __require (/var/task/netlify/functions/pricing.js:37:50)
```

Does anyone know what is wrong with this setup?

**Comment by tusbar** [comment]
Hey @kheniparth, thanks for reporting this. I’ve released a [new version](https://github.com/bizon/selling-partner-api-sdk/commit/86c909494fda76726887a6cd6819e1ded145c1df) of the packages that should [fix this issue](https://github.com/bizon/selling-partner-api-sdk/pull/546), can you please confirm?

**Comment by kheniparth** [comment]
@tusbar Thank you for the quick fix. I can confirm that the fix works and I am not getting that error anymore.

But I have started getting `InvalidClientTokenId` for some reason. This only happens while making the request from `netlify.com` domain, the same code and tokens work from the `localhost`. 

```
{
    "name": "InvalidClientTokenId",
    "$fault": "client",
    "$metadata": {
        "httpStatusCode": 403,
        "requestId": "f396536c-d92a-49de-8d88-d124526edcba",
        "attempts": 1,
        "totalRetryDelay": 0
    },
    "Type": "Sender",
    "Code": "InvalidClientTokenId",
    "message": "The security token included in the request is invalid"
}
```

Shall I create a separate issue for this?

**Comment by tusbar** [comment]
@kheniparth good to hear that the issue is fixed.

Regarding your new issue, this is probably related to AWS environment variables defined in your netlify environment that get picked up by [@sp-api-sdk/auth](https://github.com/bizon/selling-partner-api-sdk/tree/master/packages/auth#default-values-from-the-environment) that are probably not what you expect?

**Comment by kheniparth** [comment]
@tusbar I have tried hardcoding the AWS credentials instead of accessing them from environment variables as you mentioned but that didn't help.

Also, I debugged the release's env. variable by printing them to make sure everything is correct. But weren't able to make successful API calls. 

I am baffled why something like this is possible, where the same code and credentials work from localhost but fail from Netlify's subdomain. 😢 

Thank you for all the help 👍
